server/index.js

```diff
@@ -21,17 +53,35 @@ const openAIRoutes = require("./routes/openAI");
 const userProfileRoutes = require("./routes/userProfiles");
 
 // Use your authentication routes
-app.use("/api/auth", authRoutes);
+if(emailServiceEnabled){
+  app.use("/api/auth", authRoutes);
+ // Import your authentication routes
+
+}else{
+   console.log("Email service is disabled.");
+}
 
 // Use interview routes
 app.use("/api/interview", interviewRoutes);
 
+if(azureServiceEnabled){
```

Comment:
The require part for azureRoutes should also be in this if condition. Otherwise it will expect env variables even if the service is disabled. Please test thoroughly before raising the PR.